### PR TITLE
Pagination Changes

### PR DESCRIPTION
**Pull request by emmanvg**
TAXII 2.0:
- Additional handling for the Content-Range HTTP Header when provided
- Defaulting behavior to continue loop when server responds with 206 despite not providing headers

TAXII 2.1:
- Add as_pages() method to provide consistency with API.

cross-compare usage with oasis-open/cti-python-stix2#480

### CODE REVIEW

File: taxii2client/common.py [hunk 1]
```diff
@@ -130,18 +130,37 @@ def _to_json(resp):
...
+
+        results = re.match(r"^items (\d+)-(\d+)/\*$", resp.headers["Content-Range"])
+        if results:
+            return int(results.group(2)) - int(results.group(1)) + 1, float("inf")
+
+        results = re.match(r"^items \*/\*$", resp.headers["Content-Range"])
+        if results:
+            return float("inf"), float("inf")
+
+        results = re.match(r"^items \*/(\d+)$", resp.headers["Content-Range"])
```
Comment (clenk) [inline taxii2client/common.py:154]: We are assuming the "unit" of the range header is `items` but does the TAXII spec require this? Should we accept other units?

I also noticed we don't have any tests for `_grab_total_items()` or `as_pages()`.

Comment (emmanvg) [reply taxii2client/common.py:154]: Yes, TAXII 2.0 defines `items` as the range unit in section 3.4 and its the mandatory to implement unit. We shouldn't accept any other unit.

Comment (emmanvg) [reply taxii2client/common.py:154]: We don't have any tests yet, I wanted someone to comment on the new behavior for the pagination requests (aside from more support) for both specs.

Comment (emmanvg) [reply taxii2client/common.py:155]: added tests for both v20 and v21 `as_pages()`

**Review by clenk — APPROVED** [review]
LGTM; thanks @emmanvg!

**Pull request merged**